src/common/mem.c

```diff
@@ -3143,13 +3181,15 @@ cache_evict_page(struct umem_cache *cache, bool for_sys)
 static inline bool
 need_reserve(struct umem_cache *cache, uint32_t extra_pgs)
 {
-	uint32_t	page_nr;
+	uint32_t	page_nr = 0;
 
-	/* Few free pages are always reserved for potential non-evictable zone grow */
-	D_ASSERT(cache->ca_pgs_stats[UMEM_PG_STATS_NONEVICTABLE] <= cache->ca_max_ne_pages);
-	page_nr = cache->ca_max_ne_pages - cache->ca_pgs_stats[UMEM_PG_STATS_NONEVICTABLE];
-	if (page_nr > UMEM_CACHE_RSRVD_PAGES)
-		page_nr = UMEM_CACHE_RSRVD_PAGES;
+	if (!cache->ca_early_boot) {
```

Comment:
If we rename the 'ca_early_boot' to 'ca_replay_done' and set the 'ca_replay_done' flag in umem_cache_post_replay(), the new API of umem_cache_set_early_boot() could be removed.

Reply:
Fixed